src/Cake.Core/IO/DirectoryPath.cs

```diff
@@ -51,6 +52,16 @@ public FilePath GetFilePath(FilePath path)
             return new FilePath(PathHelper.Combine(FullPath, path.GetFilename().FullPath));
         }
 
+        /// <summary>
+        /// Gets the directory path of a <see cref="DirectoryPath"/>.
+        /// </summary>
+        /// <param name="path">The path.</param>
+        /// <returns>A <see cref="DirectoryPath"/> to the parent directory of the given <see cref="DirectoryPath"/>.</returns>
+        public DirectoryPath GetParent()
+        {
+            return new DirectoryPath(new DirectoryInfo(FullPath).Parent.FullName);
```

Comment:
I'm not sure we want to use .NET's `DirectoryInfo` here. We have our own custom [`PathHelper`](https://github.com/cake-build/cake/blob/3af6341ff40357282244beb59d3960c29cd6c83c/src/Cake.Core/IO/PathHelper.cs#L7) for handling paths.

@cake-build/cake-team What do you think?

Reply:
I tested the suggested code, and it also collapses the input if it was relative. Is this the desired behavior?